rackcity/views/asset_views.py

```diff
@@ -1026,9 +1029,52 @@ def get_asset_from_barcode(request):
     corresponding live asset with that number, if it exists
     """
     data = JSONParser().parse(request)
-    if "img_string" in data:
-        return JsonResponse({"img_string": data.img_string}, status=HTTPStatus.OK)
-    else:
+    if "img_string" not in data:
+        return JsonResponse(
+            {
+                "failure_message": "ERROR: Request must contain base64-encoded image string"
+            },
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    img = b64decode(data["img_string"])
+    img_file = BytesIO(img)
+    image = Image.open(img_file).convert("RGB")
+    opencv_img = numpy.array(image)
+    opencv_img = opencv_img[:, :, ::-1].copy()
+    barcodes = decode(opencv_img)
+    if len(barcodes) < 1:
+        return JsonResponse(
+            {"warning_message": "WARNING: No barcode detected"},
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    barcode_data = ""
+    try:
+        for barcode in barcodes:
+            barcode_data = barcode.data.decode("utf-8")
+    except Exception as error:
         return JsonResponse(
-            {"failure_message": "failed"}, status=HTTPStatus.BAD_REQUEST
+            {"failure_message": "ERROR: Could not decode barcode value"},
+            status=HTTPStatus.BAD_REQUEST,
         )
+    if not barcode_data.isdigit():
+        return JsonResponse(
+            {"failure_message": "ERROR: Barcode contains non-numeric characters"},
+            status=HTTPStatus.BAD_REQUEST,
+        )
+    try:
+        asset = Asset.objects.get(asset_number=barcode_data)
+    except Asset.DoesNotExist:
+        return JsonResponse(
+            {
+                "failure_message": Status.ERROR.value
+                + "Asset"
+                + GenericFailure.DOES_NOT_EXIST.value,
+                "errors": "No existing asset with asset_number=" + str(barcode_data),
+            },
+            status=HTTPStatus.BAD_REQUEST,
+        )
```

Comment:
what do you think about having a more general failure message to show to the user? I feel like the differences don't matter to the user, they just need to know that there's no hyposoft asset associated with the barcode they sent - so maybe something like `"There is no existing asset associated with this barcode"`?